### PR TITLE
avoid log fatal

### PR DESCRIPTION
**Pull request by jsign**

### CODE REVIEW

File: api/pb/threads.proto [hunk 1]
```diff
@@ -20,23 +20,27 @@ message GetTokenReply {
...
-    repeated CollectionConfig collections = 3;
+    bytes logKey = 7;
     string name = 4;
+    repeated CollectionConfig collections = 3;
     bool block = 5;
-    bytes threadKey = 6;
-    bytes logKey = 7;
+
+    bytes threadKey = 6 [deprecated = true];
 }
```
Comment (sanderpick) [inline api/pb/threads.proto:44]: Somehow I missed this in Daniel's PR. `threadKey` was added, but we already have `key`. This change does not modify the proto tag numbers. We can clean this up on the next major release.

File: api/service.go [hunk 2]
```diff
@@ -191,12 +191,12 @@ func (s *Service) NewDB(ctx context.Context, req *pb.NewDBRequest) (*pb.NewDBRep
 	if _, err = s.manager.NewDB(
 		ctx,
 		id,
+		db.WithNewManagedKey(key),
+		db.WithNewManagedLogKey(logKey),
 		db.WithNewManagedName(req.Name),
+		db.WithNewManagedCollections(collections...),
 		db.WithNewManagedToken(token),
-		db.WithNewManagedBackfillBlock(req.Block),
```
Comment (sanderpick) [inline api/service.go:196]: This has no meaning in the context of `NewDB`

File: db/db.go [hunk 3]
```diff
@@ -95,24 +99,46 @@ func NewDB(ctx context.Context, network app.Net, id thread.ID, opts ...NewOption
 		opt(args)
 	}
 
-	if _, err := network.CreateThread(ctx, id, net.WithThreadKey(args.ThreadKey), net.WithLogKey(args.LogKey), net.WithNewThreadToken(args.Token)); err != nil {
-		if !errors.Is(err, lstore.ErrThreadExists) && !errors.Is(err, lstore.ErrLogExists) {
-			return nil, err
-		}
+	if args.Key.Defined() && !args.Key.CanRead() {
+		return nil, ErrThreadReadKeyRequired
```
Comment (sanderpick) [inline db/db.go:103]: Early out if thread key does not contain a read component.

**Review by sanderpick — APPROVED** [review]

**Comment by jtacoma** [comment]
FYI, this changed the public API: `db.WithNewThreadKey` was renamed to `db.WithNewKey`.

**Comment by sanderpick** [comment]
Doh, thanks for the heads up @jtacoma - I'll make a not on the release that includes those changes.

**Pull request merged**